### PR TITLE
feat: Universe Domain Environment Variable Support

### PR DESCRIPTION
**Pull request by lqiu96**
Support for reading the `GOOGLE_CLOUD_UNIVERSE_DOMAIN` env var.

Since we are still on JUnit 4 and there is no dedicated support for testing Env Vars, we created a new step in the CI to test this Env Var test:
image

### CODE REVIEW

**Comment by lqiu96** [comment]
On a second look, I think this might cause issues when running this locally. This test will fail if the local environment does not have the `GOOGLE_CLOUD_UNIVERSE_DOMAIN` env var set. I think I'll change this to a maven profile that needs to be activated (and is disabled by default).

File: gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java [hunk 1]
```diff
@@ -283,6 +285,11 @@ String mtlsEndpointResolver(
     }
 
     public EndpointContext build() throws IOException {
+      // If the universe domain wasn't configured explicitly in the settings, check the
```
Comment (blakeli0) [inline gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java:288]: Can we move the logic to `determineUniverseDomain()`?

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java [hunk 2]
```diff
@@ -340,6 +340,21 @@ public void endpointContextBuild_gdchFlow_noUniverseDomain_customEndpoint() thro
         .isEqualTo(Credentials.GOOGLE_DEFAULT_UNIVERSE);
   }
 
+  // This Universe Domain should match the `GOOGLE_CLOUD_UNIVERSE_DOMAIN` Env Var
+  // For this test running locally or in CI, check that the Env Var is set properly.
+  // This test should only run when the maven profile `EnvVarTest` is enabled.
+  @Test
+  public void endpointContextBuild_universeDomainEnvVarSet() throws IOException {
```
Comment (blakeli0) [inline gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java:347]: Can you please add one more test case that if universe domain is set on both ClientSettings and Env Var, then ClientSettings should take precedence.

Comment (lqiu96) [reply gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java:347]: Yep, added a test case that will be run with `-PenvVarTest`

File: gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java (hunk 1 above)
Comment (lqiu96) [reply gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java]: Sure, moved it to `determineUniverseDomain()`

**Pull request merged**